packages/zealot/src/client/client.ts

```diff
@@ -0,0 +1,229 @@
+import {EventSourcePolyfill} from "event-source-polyfill"
+import nodeFetch from "node-fetch"
+import {PoolConfig, PoolStats, zjson} from ".."
+import {decode} from "../encoder"
+import {ResultStream} from "../query/result-stream"
+import {parseContentType} from "../util/content-type"
+import {createError} from "../util/error"
+import {
+  ClientOpts,
+  CreatePoolOpts,
+  CreatePoolResp,
+  CrossFetch,
+  LoadOpts,
+  Pool,
+  QueryOpts,
+  ResponseFormat
+} from "./types"
+
+export class Client {
```

Comment:
This is the new zealot client class. It's flat and contains duplication at the moment. I wanted to get it all out here like this first, then clean up the duplication.